### PR TITLE
[ibex/doc] initial ibex checklist

### PR DESCRIPTION
**Pull request by udinator**
Creating this PR to address #773 - as mentioned in that discussion, we are assigning Ibex V1 status for right now, but we might need to slightly alter this checklist format to more directly address Ibex, since it doesn't follow any of the comportability guidelines.

### CODE REVIEW

File: hw/ip/rv_core_ibex/doc/checklist.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+Documentation | [BLOCK_DIAGRAM][]       | Done        |
+Documentation | [DOC_INTERFACE][]       | Done        |
+Documentation | [MISSING_FUNC][]        | N/A         |
+Documentation | [FEATURE_FROZEN][]      | Done        |
+RTL           | [FEATURE_COMPLETE][]    | Done        |
+RTL           | [AREA_SANITY_CHECK][]   | Done        | Area Sanity Check Done (on FPGA)
+RTL           | [PORT_FROZEN][]         | Done        |
+RTL           | [ARCHITECTURE_FROZEN][] | Done        |
+RTL           | [REVIEW_TODO][]         | Done        | Minor TODOs remain, waived
+RTL           | [STYLE_X][]             | N/A         | No assignment of X
```
[the remark this answers is not in the document]
Comment (udinator) [reply hw/ip/rv_core_ibex/doc/checklist.md]: Shall we mark this as `Done` then, with a note referring to #366?

File: hw/ip/rv_core_ibex/doc/checklist.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+
+## Design Checklist
+
+### D1
+
+Type          | Item                  | Resolution  | Note/Collaterals
+--------------|-----------------------|-------------|------------------
+Documentation | [SPEC_COMPLETE][]     | Done        |
+Documentation | [CSR_DEFINED][]       | Done        |
```
[the remark this answers is not in the document]
Comment (udinator) [reply hw/ip/rv_core_ibex/doc/checklist.md]: Sure, I will add a note here about this issue. Thanks for pointing that out!

**Comment by udinator** [comment]
@vogelpi - thanks for the comments!
@all - this PR got killed after I forked off the new OT repository, forcing me to create a new PR for this checklist, it is available at #872.

**Comment by imphil** [comment]
Merged the updated version https://github.com/lowRISC/opentitan/pull/872 instead.

**Pull request closed without merging**